api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -34,117 +42,109 @@ public Cluster apply(org.pmiops.workbench.notebooks.model.Cluster firecloudClust
+    } catch (NotFoundException e) {
+      fcCluster = this.notebooksService.createCluster(project, clusterName, createFirecloudClusterRequest());
+    }
+    ClusterListResponse resp = new ClusterListResponse();
+    resp.setDefaultCluster(TO_ALL_OF_US_CLUSTER.apply(fcCluster));
+    return ResponseEntity.ok(resp);
   }
 
-
-  @Override
-  public ResponseEntity<ClusterListResponse> listClusters(String labels) {
-    List<org.pmiops.workbench.notebooks.model.Cluster> oldClusters;
-    oldClusters = this.notebooksService.listClusters(labels, /* includeDeleted */ false);
-    ClusterListResponse response = new ClusterListResponse();
-    response.setItems(oldClusters.stream().map(TO_ALL_OF_US_CLUSTER).collect(Collectors.toList()));
-    return ResponseEntity.ok(response);
+  private org.pmiops.workbench.notebooks.model.ClusterRequest createFirecloudClusterRequest() {
+    org.pmiops.workbench.notebooks.model.ClusterRequest firecloudClusterRequest = new org.pmiops.workbench.notebooks.model.ClusterRequest();
+    // TODO(calbach): Configure Jupyter server extensions here.
+    Map<String, String> labels = new HashMap<String, String>();
+    labels.put("all-of-us", "true");
+    labels.put("created-by", userProvider.get().getEmail());
+    firecloudClusterRequest.setLabels(labels);
+    return firecloudClusterRequest;
   }
 
   @Override
-  public ResponseEntity<Void> localizeNotebook(String workspaceNamespace, String workspaceId,
-      List<FileDetail> fileList) {
-    String clusterName = convertClusterName(workspaceId);
-    this.notebooksService.localize(workspaceNamespace, clusterName,
-        convertfileDetailsToMap(fileList));
+  public ResponseEntity<Void> localize(
+      String projectName, String clusterName, ClusterLocalizeRequest body) {
+    String workspaceBucket;
+    try {
+      workspaceBucket = "gs://" + fireCloudService.getWorkspace(body.getWorkspaceNamespace(), body.getWorkspaceId())
+          .getWorkspace()
+          .getBucketName();
+    } catch (ApiException e) {
+      if (e.getCode() == 404) {
+        log.log(Level.INFO, "Firecloud workspace not found", e);
+        throw new NotFoundException(String.format(
+            "workspace %s/%s not found or not accessible",
+            body.getWorkspaceNamespace(), body.getWorkspaceId()));
+      }
+      throw ExceptionUtils.convertFirecloudException(e);
+    }
+
+    // For the common case where the notebook cluster matches the workspace
+    // namespace, simply name the directory as the workspace ID; else we
+    // include the namespace in the directory name to avoid possible conflicts
+    // in workspace IDs.
+    String workspacePath = body.getWorkspaceId();
+    if (!projectName.equals(body.getWorkspaceNamespace())) {
+      workspacePath = body.getWorkspaceNamespace() + ":" + body.getWorkspaceId();
+    }
+    String localDir = String.join("/", "~", "workspaces", workspacePath);
+    Map<String, String> toLocalize = body.getNotebookNames()
```

Comment:
So this still localizes all notebooks, right?

I'm wondering if there's a way to do this initial localization with initialization scripts, though I get stuck when it comes to passing in parameters (e.g. which bucket to grab stuff from.)